torchgeo/samplers/utils.py

```diff
@@ -4,13 +4,23 @@
 """Common sampler utilities."""
 
 import math
-from typing import Optional, Tuple, Union
+from typing import Optional, Tuple, Union, overload
 
 import torch
 
 from ..datasets import BoundingBox
 
 
+@overload
+def _to_tuple(value: Union[Tuple[int, int], int]) -> Tuple[int, int]:
+    ...
+
+
+@overload
+def _to_tuple(value: Union[Tuple[float, float], float]) -> Tuple[float, float]:
+    ...
```

Comment:
In Python typing, all ints are floats, but not all floats are ints. This meant that if I pass an int as input, mypy would consider its output type to be float. These overloads ensure that int maps to int and float maps to float as expected.